docs/DeveloperGuide.md

```diff
@@ -614,3 +614,92 @@ testers are expected to do more *exploratory* testing.
+       Expected: No changes. UGTeach displays error message of invalid command format.
+
+### Using owe command
+
+1. Using owe command while all students are being shown.
+
+    1. Prerequisite: List all students using the `list` command. There should be **at least 1 student** listed.
+
+    1. Test case: `owe 1 hr/1`<br>
+       Expected: Owed amount for 1st student increases by (1 hr * rate). UGTeach displays a message that 1st student owed another (1 hr * rate).
+
+    1. Test case: `owe 1 hr/-1`<br>
+       Expected: No changes. UGTeach displays error message asking for positive multiple of 0.5 when inputting hours.
+
+    1. Test case: `owe 0 hr/1`<br>
+       Expected: No changes. UGTeach displays error message of invalid command format.
+
+1. Using owe command from a filtered list.
+
+    1. Prerequisite: Find a student using the `find` command. There should be **at least 1 student** found.
+
+    1. Test case: `owe 1 hr/1`<br>
+       Expected: Owed amount for 1st student in **filtered list** increases by (1 hr * rate). UGTeach displays a message that 1st student owed another (1 hr * rate).
+
+    1. Test case: `owe 1 hr/-1`<br>
+       Expected: No changes. UGTeach displays error message asking for positive multiple of 0.5 when inputting hours.
+
+    1. Test case: `owe 0 hr/1`<br>
+       Expected: No changes. UGTeach displays error message of invalid command format.
+
+### Using settle command
+
+1. Using settle command while all students are being shown.
+
+    1. Prerequisite: List all students using the `list` command. There should be **at least 1 student** listed.
+
+    1. Test case: `settle 1 amount/10`<br>
+       Expected: For 1st student, assuming owed amount is more than 10, owed amount decreases by 10, while paid amount increases by 10. UGTeach displays a message that payment of 10.00 has been settled.
+
+    1. Test case: `settle 1 amount/-10`<br>
+       Expected: No changes. UGTeach displays error message that amount needs to be positive.
+
+    1. Test case: `settle 0 amount/10`<br>
+       Expected: No changes. UGTeach displays error message of invalid command format.
+
+    1. Test case: `settle 1 amount/10000`<br>
+       Expected: No changes, assuming that the amount entered is more than the amount owed.
+       UGTeach displays error message that entered amount is more than amount owed.
+
+1. Using settle command from a filtered list.
+
+    1. Prerequisite: Find a student using the `find` command. There should be **at least 1 student** found.
+
+    1. Test case: `settle 1 amount/10`<br>
+       Expected: For 1st student in **filtered list**, assuming owed amount is more than 10, owed amount decreases by 10, while paid amount increases by 10. UGTeach displays a message that payment of 10.00 has been settled.
+
+    1. Test case: `settle 1 amount/-10`<br>
+       Expected: No changes. UGTeach displays error message that amount needs to be at most 2 decimal places and must be between
```

Comment:
I don't see the need to elaborate all these error messages. Your line 682 and line 699 seems to test for the same thing but expected is different.
`Displays an error message` is enough.

Reply:
I think he forgot to change the expected for line 682. But I think can perhaps say 'displays error message notifying that amount is invalid' for where ever applicable

Reply:
Sounds good